axelrod/fingerprint.py

```diff
@@ -0,0 +1,206 @@
+            produce more coordinates that will be closer together.
+        cores : integer, optional
+            The number of processes to be used for parallel processing
+        """
+        self.probe_players = self.create_probes(self.probe, granularity)
+        self.edges = self.create_edges(self.probe_players)
+        original = self.strategy()
+        dual = DualTransformer()(self.strategy)()
+        probes = self.probe_players.values()
+        tourn_players = [original, dual] + list(probes)
+        spatial_tourn = axl.SpatialTournament(tourn_players, turns=turns,
+                                              repetitions=repetitions,
+                                              edges=self.edges)
+        print("Begin Spatial Tournament")
+        self.results = spatial_tourn.play(processes=cores,
+                                          keep_interactions=True)
+        print("Spatial Tournament Finished")
+
+    def _generate_data(self, results, probe_coords):
+        """Generates useful data from a spatial tournament.
+
+        Matches interactions from `results` to their corresponding coordinate in
+        `probe_coords`.
+
+        Parameters
+        ----------
+        results : axelrod.result_set.ResultSetFromFile
+            A results set for a spatial tournament.
+        probe_coords : list of tuples
+            A list of tuples of length 2, where each tuple represents a
+            coordinate, eg. (x, y).
+
+        Returns
+        ----------
+        data_frame : pandas.core.frame.DataFrame
+            A pandas DataFrame object where the row and column headers
+            correspond to coordinates. The cell values are the score of the
+            original/dual strategy playing the probe with parameters that match
+            the coordinate.
+        """
+        edge_scores = {key: cfspt(value[0])[0] for key, value in
+                       results.interactions.items()}
+
+        coord_scores = OrderedDict.fromkeys(probe_coords)
+        for index, coord in enumerate(coord_scores.keys()):
+            if sum(coord) > 1:
+                edge = (1, index + 2)
+            else:
+                edge = (0, index + 2)
+            coord_scores[coord] = edge_scores[edge]
+
+        ser = pd.Series(list(coord_scores.values()),
+                        index=pd.MultiIndex.from_tuples(coord_scores.keys()))
+        data_frame = ser.unstack().fillna(0)
+        data_frame.shape
+        return data_frame
+
+    def plot(self, col_map=None):
+        """Plot the results of the spatial tournament.
+
+        Parameters
+        ----------
+        col_map : str, optional
+            A matplotlib colour map, full list can be found at
+            http://matplotlib.org/examples/color/colormaps_reference.html
+        """
+        self.data = self._generate_data(self.results, self.probe_players.keys())
+        sns.heatmap(self.data, cmap=col_map)
+        plt.show()
```

Comment:
I wonder if some of the methods in here might be better as module level functions. They could then be tested in complete isolation without the need to create a fingerprint instance.